delete-tags.js

```diff
@@ -0,0 +1,49 @@
+module.exports = async ({ github, context, core }) => {
+  const { IMAGE_NAME, ORGANIZATION } = process.env
+
+  let tagName
+
+  if (context.payload.ref_type === 'branch') {
+    tagName = `branch-${context.payload.ref}`
+  } else {
+    tagName = context.payload.ref.match(/^v(.*)$/)[1]
+  }
+
+  let didDelete = false
+
+  for await (const response of github.paginate.iterator(
+    github.rest.packages.getAllPackageVersionsForPackageOwnedByOrg, {
+      package_type: 'container',
+      package_name: IMAGE_NAME,
+      org: ORGANIZATION
+    })) {
+    for (const version of response.data) {
+      const tags = version.metadata?.container?.tags
+      if (tags?.includes(tagName)) {
+        core.notice(`Package version ${version.html_url} matches tag ${tagName} and will be deleted`)
+
+        const otherTags = tags.filter((tag) => tag !== tagName)
+        if (otherTags.length) {
+          core.warning(`Image version has other tags that will be lost: ${otherTags}`)
+        }
+
+        await github.rest.packages.deletePackageVersionForOrg({
+          package_type: 'container',
+          package_name: IMAGE_NAME,
+          org: ORGANIZATION,
+          package_version_id: version.id
+        })
+
+        didDelete = true
+        break
```

Comment:
**question (blocking):**

Will `deletePackageVersionForOrg` delete all items that exist in `response.data`?

Reply:
No, it deletes at most one specific image digest (e.g. `sha256:4cd058b01e9aea1d13a6a8a403045bf7b7f9e2f162aee6a5e696a4efab885e30`) referenced by the tag of interest.

If there are multiple tags pointing to the exact same image (they'll be listed under "Other tags on this version" on the [version's page](https://github.com/uclahs-cds/docker-internal-tests/pkgs/container/internal-tests/234407984?tag=dev)) then those _tags_ will also be deleted, but it's still only one image version under the hood.

Reply:
As discussed, branches are prefixed with `branch-` and in general the lab's configurations will prevent this kind of ambiguity. No need for a change here.